packages/cdktf-cli/bin/cmds/ui/models/terraform.ts

```diff
@@ -54,84 +49,17 @@ export interface TerraformOutput {
   value: string;
 }
 
-export class TerraformPlan {
-  constructor(public readonly planFile: string, public readonly plan: {[key: string]: any}) {}
-
-  public get resources(): PlannedResource[]  {
-    if (!this.plan.resource_changes) return [];
-
-    return this.plan.resource_changes.map((resource: ResourceChanges) => {
-      return {
-        id: resource.address,
-        action: resource.change.actions[0]
-      } as PlannedResource
-    })
-  }
-
-  public get applyableResources(): PlannedResource[] {
-    const applyActions = [PlannedResourceAction.UPDATE, PlannedResourceAction.CREATE, PlannedResourceAction.DELETE, PlannedResourceAction.READ];
-    return this.resources.filter(resource => (applyActions.includes(resource.action)));
-  }
-
-  public get needsApply(): boolean {
-    return this.applyableResources.length > 0
-  }
+export interface TerraformPlan {
```

Comment:
Currently, there are subtile differences in the implementation (e.g. camel cased in the cloud context). And since the Cloud story isn't fully complete with this PR, I wouldn't necessarily assume these are the only differences. However, later on, it could certainly make sense to extract parts of the implementation into an abstract class.